flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/CodeGeneratorContext.scala

```diff
@@ -603,7 +603,7 @@ class CodeGeneratorContext(val tableConfig: TableConfig) {
     val byteArray = InstantiationUtil.serializeObject(obj)
     val objCopy: AnyRef = InstantiationUtil.deserializeObject(
       byteArray,
-      obj.getClass.getClassLoader)
+      Thread.currentThread().getContextClassLoader)
     references += objCopy
```

Comment:
Can we give a test case to prove that the `Thread.currentThread().getContextClassLoader` is the right one to choose ?

Reply:
The proper place to put the test code is module `flink-scala-shell`. But after some investigation, I just found that it is not easy to do that because currently scala shell doesn't' support switch planner. I may need to create another PR to support that, but that seems too complicated. 
Since this is a straightforward fix, I think it is OK without test code.  What do you think @danny0405

Reply:
I'm fine with this, there seems some compile error.